src/Workspaces/Core/Portable/Shared/Extensions/INamespaceSymbolExtensions.cs

```diff
@@ -79,6 +79,25 @@ public static int CompareTo(this INamespaceSymbol n1, INamespaceSymbol n2)
             }
         }
 
+        public static IEnumerable<INamespaceSymbol> GetAllNamespaces(
+            this INamespaceSymbol namespaceSymbol,
+            CancellationToken cancellationToken)
+        {
+            var stack = new Stack<INamespaceSymbol>();
```

Comment:
imo that will just make it messier than it currently is, since there's enough small differences. I'm open to suggestions if you know a good unifying pattern.

Reply:
Talked offline, agreed current is fine